### PR TITLE
feat(gallery): added max variable

### LINKED ISSUE #3701: Enhancement: add max-width var to Gallery
I was using the `gallery` layout in product and pretty quickly realized that having a `--pf-l-gallery--GridTemplateColumns--max{-on-[breakpoint]}: {width}` would complete the layout customization.

### PR DESCRIPTION
**Pull request by mattnolting**
closes #3701

### CODE REVIEW

File: src/patternfly/layouts/Gallery/examples/Gallery.md [hunk 1]
```diff
@@ -55,6 +66,28 @@ Grid template columms min value can be applied to <code>.pf-l-gallery</code>.  M
...
+  {{#> gallery-item}}content{{/gallery-item}}
+  {{#> gallery-item}}content{{/gallery-item}}
+  {{#> gallery-item}}content{{/gallery-item}}
+  {{#> gallery-item}}content{{/gallery-item}}
+{{/gallery}}
+```
+
+### Min and max width modified gallery, responsive updates
+```hbs
+{{#> gallery gallery--modifier="pf-m-gutter" gallery--attribute='style="--pf-l-gallery--GridTemplateColumns--min: 100%; --pf-l-gallery--GridTemplateColumns--min-on-md: 100px; --pf-l-gallery--GridTemplateColumns--max-on-md: 200px; --pf-l-gallery--GridTemplateColumns--min: 100%; --pf-l-gallery--GridTemplateColumns--min-on-xl: 300px; --pf-l-gallery--GridTemplateColumns--max-on-xl: 1fr;"'}}
```
Comment (mcoker) [inline src/patternfly/layouts/Gallery/examples/Gallery.md:82]:
```suggestion
{{#> gallery gallery--modifier="pf-m-gutter" gallery--attribute='style="--pf-l-gallery--GridTemplateColumns--min: 100%; --pf-l-gallery--GridTemplateColumns--min-on-md: 100px; --pf-l-gallery--GridTemplateColumns--max-on-md: 200px; --pf-l-gallery--GridTemplateColumns--min-on-xl: 300px; --pf-l-gallery--GridTemplateColumns--max-on-xl: 1fr;"'}}
```

**Review by mcoker — CHANGES_REQUESTED** [review]
Looks great! You have `--pf-l-gallery--GridTemplateColumns--min: 100%;` twice in the min/max/responsive example though.

File: src/patternfly/layouts/Gallery/examples/Gallery.md (hunk 1 above)
Comment (mattnolting) [reply src/patternfly/layouts/Gallery/examples/Gallery.md:82]: doh!

**Review by mcoker — APPROVED** [review]
👍👍

**Pull request merged**